crates/biome_js_analyze/src/lint/nursery/use_iterable_callback_return.rs

```diff
@@ -0,0 +1,378 @@
+use crate::ControlFlowGraph;
+use crate::services::control_flow::JsControlFlowGraph;
+use biome_analyze::{Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule};
+use biome_console::fmt::{Display, Formatter};
+use biome_console::markup;
+use biome_control_flow::builder::ROOT_BLOCK_ID;
+use biome_control_flow::{ExceptionHandlerKind, InstructionKind};
+use biome_js_syntax::{
+    AnyJsFunctionBody, JsArrowFunctionExpression, JsCallArgumentList, JsCallArguments,
+    JsCallExpression, JsFunctionExpression, JsReturnStatement, global_identifier,
+};
+use biome_rowan::{AstNode, AstSeparatedList, NodeOrToken, TextRange};
+use roaring::RoaringBitmap;
+use rustc_hash::FxHashMap;
+use std::io;
+use std::sync::LazyLock;
+
+declare_lint_rule! {
+    /// Enforce consistent return values in iterable callbacks.
+    ///
+    /// This rule ensures that callbacks passed to certain iterable methods either always return a value or never return a value, depending on the method's requirements.
+    ///
+    /// ## Methods and Their Requirements
+    ///
+    /// - Methods like `map`, `filter`, `reduce`, `some`, `every`, etc., require the callback to always return a value.
+    /// - Methods like `forEach` do not require the callback to return a value.
```

Comment:
Since we are teaching users things, we should list **all** methods, so they don't need to go to another website (which we don't link) to learn about the methods that require a `return` in their callback.

So the phrase like "Methods like" should become "The following methods require a return in their callback: `map`, [...the rest of the methods]"

Reply:
Done